matrix/lambdas/api/v1/core.py

```diff
@@ -51,6 +51,10 @@ def post_matrix(body: dict):
         # Otherwise, default to human-only
         genera_species = [constants.GenusSpecies.HUMAN]
 
+    # Cell barcode is required in .mtx matrices by 10x specifications
+    if format_ == MatrixFormat.MTX.value and "cell.barcode" not in fields:
```

Comment:
I think you want `"cell.barcode" not in fields and "barcode" not in fields`. If it's the latter, it will get translated to `cell.barcode` in the query_constructor.

Reply:
Oh I see. I thought `cell.barcode` was the user-facing field since it's the default name used in both APIs, but `barcode` is used everywhere else (`filters` endpoint, queries). I believe we want to just stop using `cell.barcode` everywhere. Note that most matrices' metadata have been generated with `cell.barcode` and this may break downstream analyses.

Reply:
Nevermind, I see what you're saying re: `query_constructor._get_internal_name`. Going with suggestion